File: snippets/csharp/ConnectingLsp.cs
using Breez.Sdk;

public class ConnectingLspSnippets
{
    public void GetLspInfo(BlockingBreezServices sdk)
    {
        // ANCHOR: get-lsp-info
        try
        {
            var lspId = sdk.LspId();
            var lspInfo = sdk.LspInfo();
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: get-lsp-info
    }

    public void ListLsps(BlockingBreezServices sdk)
    {
        // ANCHOR: list-lsps
        try
        {
            var availableLsps = sdk.ListLsps();
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-lsps
    }

    public void ConnectLsp(BlockingBreezServices sdk, string? lspId)
    {
        // ANCHOR: connect-lsp
        try
        {
            sdk.ConnectLsp(lspId!);
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: connect-lsp
    }
}

File: snippets/python/src/connecting_lsp.py

def get_lsp_info(sdk_services):
    try: 
        # ANCHOR: get-lsp-info
        lsp_id = sdk_services.lsp_id()
        lsp_info = sdk_services.lsp_info()
        # ANCHOR_END: get-lsp-info
        return lsp_info
    except Exception as error: 
        print(error)
        raise

def list_lsps(sdk_services):
    try:
        # ANCHOR: list-lsps
        available_lsps = sdk_services.list_lsps()
        # ANCHOR_END: list-lsps
    except Exception as error:
        print(error)
        raise

def connect_lsp(sdk_services,lsp_id):
    try:
        # ANCHOR: connect-lsp
        sdk_services.connect_lsp(lsp_id)
        # ANCHOR_END: connect-lsp
    except Exception as error:
        print(error)
        raise